jdaviz/configs/mosviz/plugins/parsers.py

```diff
@@ -87,6 +89,32 @@ def _fields_from_ecsv(fname, fields, delimiter=","):
     return parsed_fields
 
 
+@data_parser_registry("mosviz-link-data")
+def link_data_in_table(app, data_obj=None, data_labels=None):
+    """
+    Batch links data in the mosviz table viewer
+
+    Parameters
+    ----------
+    app : `~jdaviz.app.Application`
+        The application-level object used to reference the viewers.
+    data_obj : None
+        Passed in in order to use the data_parser_registry, otherwise
+        not used.
+    data_labels : None
+        Not used.
+    """
+    mos_data = app.session.data_collection['MOS Table']
+
+    # Loop through mos table rows and link Spectra 1D and Spectra 2D
+    # data objects. This speeds up performance significantly over linking
+    # data when loaded into the application.
+    for index in range(0, len(mos_data.get_component('1D Spectra').data)):
+        spec_1d = mos_data.get_component('1D Spectra').data[index]
+        spec_2d = mos_data.get_component('2D Spectra').data[index]
+        app.session.data_collection.add_link(LinkSame(spec_1d, spec_2d))
```

Comment:
Is there a need to check if that link already exists before actually adding it?

Reply:
I'm not sure, I can look into how to do that though. I think we can assume that someone using Mosviz will have `auto_link` set to False since it speeds up data loading considerably, but you know what they say about assumptions...